prow/pjutil/pjutil.go

```diff
@@ -50,9 +48,18 @@ func NewProwJob(spec kube.ProwJobSpec, labels map[string]string) kube.ProwJob {
 }
 
 func newProwJob(spec kube.ProwJobSpec, labels, annotations map[string]string) kube.ProwJob {
+	jobNameForLabel := spec.Job
+	if len(jobNameForLabel) > validation.LabelValueMaxLength {
```

Comment:
It's possible that a simple truncation will create an invalid label value -- for instance, if the last char of the new label value ends in `-`. We should continue truncating characters past 63 to achieve a valid label

Reply:
are you aware of an upstream API we can consume for this kind of resource validation? I've had similar issues with `kind` and wound up rolling my own more-restrictive validation thus far (it's a value overloaded for other uses in this case anyhow), but probably we have a lot more use cases for this sort of thing... :thinking: